storage/ctab/src/main/java/org/openscience/cdk/io/MDLV3000Writer.java

```diff
@@ -513,18 +513,22 @@ private IAtom[] pushHydrogensToBack(IAtomContainer mol, Map<IChemObject, Integer
             throw new AssertionError("Map atomToIdx must be empty.");
 
         final IAtom[] atoms = new IAtom[mol.getAtomCount()];
+        final List<IAtom> hydrogenAtoms = new ArrayList<>();
```

Comment:
Why was this needed? It's fine but actually the second copy array is marginally less efficient - but in practice it won't mater. Technically a dutch national flag like algorithm would be best but this function isn't critical.

Reply:
I apologize, that wasn't the best of changes. I changed the implementation in `MDLV3000Writer::pushHydrogensToBack` in commit [fc06c89](https://github.com/cdk/cdk/pull/1106/commits/fc06c89c4bfa66949e06e06dcf43c4dfe40e351c) to something that I would consider along the lines of a 2 way partitioning algorithm. But happy to go back on that change if you don't like it.